src/databricks/labs/ucx/hive_metastore/mapping.py

```diff
@@ -103,14 +114,82 @@ def skip_table(self, backend: StatementExecutionBackend, schema: str, table: str
+        if self.UCX_SKIP_PROPERTY in TablesCrawler.parse_database_props(describe.get("Properties", "").lower()):
+            logger.info(f"Database {database} is marked to be skipped")
+            return None
+        return database
+
+    def get_tables_to_migrate(self, tables_crawler: TablesCrawler):
+        rules = self.load()
+        # Getting all the source tables from the rules
+        databases_in_scope = self._get_databases_in_scope({rule.src_schema for rule in rules})
+        crawled_tables_keys = {crawled_table.key: crawled_table for crawled_table in tables_crawler.snapshot()}
+        tasks = []
+        for rule in rules:
+            if rule.as_hms_table_key not in crawled_tables_keys:
+                logger.info(f"Table {rule.as_hms_table_key} in the mapping doesn't show up in assessment")
+                continue
+            if rule.src_schema not in databases_in_scope:
+                logger.info(f"Table {rule.as_hms_table_key} is in a database that was marked to be skipped")
+                continue
+            tasks.append(
+                partial(self._get_table_in_scope_task, TableToMigrate(crawled_tables_keys[rule.as_hms_table_key], rule))
+            )
+
+        return Threads.strict("checking all database properties", tasks)
+
+    def _get_table_in_scope_task(self, table_to_migrate: TableToMigrate) -> TableToMigrate | None:
+        table = table_to_migrate.src
+        rule = table_to_migrate.rule
+
+        if self._exists_in_uc(rule.as_uc_table_key):
+            logger.info(f"The intended target for {table.key}, {rule.as_uc_table_key}, already exists.")
+            return None
+        result = self._backend.fetch(f"SHOW TBLPROPERTIES `{table.database}`.`{table.name}`")
+        for value in result:
+            if value["key"] == self.UCX_SKIP_PROPERTY:
+                logger.info(f"{table.key} is marked to be skipped")
+                return None
+            if value["key"] == "upgraded_to":
+                logger.info(f"{table.key} is set as upgraded to {value['value']}")
+                if self._exists_in_uc(value["value"]):
+                    logger.info(
+                        f"The table {table.key} was previously upgraded to {value['value']}. "
+                        f"To revert the table and allow it to be upgraded again use the CLI command:"
+                        f"databricks labs ucx revert --schema {table.database} --table {table.name}"
+                    )
+                    return None
+                logger.info(f"The upgrade_to target for {table.key} is missing. Unsetting the upgrade_to property")
+                self._backend.execute(table.sql_unset_upgraded_to())
+
+        return table_to_migrate
+
+    def _exists_in_uc(self, target_key: str):
+        # Attempts to get the target table info from UC returns True if it exists.
+        try:
+            self._ws.tables.get(target_key)
```

Comment:
Get upgraded_from property from result and compare it with the source table. Throw exception if it doesn't match

Reply:
Fixed